watertrading/code/MainScript.py

```diff
@@ -0,0 +1,53 @@
+##################################################
+# PYTHON IMPORTS
+import json
+import pandas as pd
+from datetime import datetime, timedelta
+from pyomo.opt import TerminationCondition
+import pyomo.environ
+from pyomo.environ import value
+from pyomo.opt import SolverFactory
+import BuildModel
+from os import getcwd
+
+
+##################################################
+# RUN OPTIMIZATION ALGORITHM
+def run_optimization_models(REQUESTS,DISTANCE,MATCHES,PROFITS,Update_distance_matrix=True,save_dir=getcwd(),verbose=True,OnlineMode=False,filter_ON=False,filter_date=None):
```

Comment:
Another comment, as mentioned above, it is not fine having so many args, instead pass them in a dictionary. It is very common to mess the order up and not realize. Another thing that I saw with this code is that there are no error checks, so it will be really hard to figure exactly which ones are mixed up unless otherwise analyzed